homeassistant/components/google_assistant/trait.py

```diff
@@ -1397,7 +1397,7 @@ def __init__(self, hass, state, config):
         if state.domain == fan.DOMAIN:
             speed_count = min(
                 FAN_SPEED_MAX_SPEED_COUNT,
-                round(100 / self.state.attributes.get(fan.ATTR_PERCENTAGE_STEP) or 1.0),
+                round(100 / self.state.attributes.get(fan.ATTR_PERCENTAGE_STEP, 100)),
```

Comment:
```suggestion
                round(100 / (self.state.attributes.get(fan.ATTR_PERCENTAGE_STEP) or 1.0)),
```

That was what the or 1.0 was for. But it must be a operator precedence issue. It just needs some (). I explicitly didnt use the default of the get since the attribute could be None instead of missing